Library/Homebrew/cmd/upgrade.rb

```diff
@@ -286,21 +295,28 @@ def upgrade_outdated_casks!(casks)
 
         Install.ask_casks casks if args.ask?
 
-        Cask::Upgrade.upgrade_casks!(
-          *casks,
-          force:               args.force?,
-          greedy:              args.greedy?,
-          greedy_latest:       args.greedy_latest?,
-          greedy_auto_updates: args.greedy_auto_updates?,
-          dry_run:             args.dry_run?,
-          binaries:            args.binaries?,
-          quarantine:          args.quarantine?,
-          require_sha:         args.require_sha?,
-          skip_cask_deps:      args.skip_cask_deps?,
-          verbose:             args.verbose?,
-          quiet:               args.quiet?,
-          args:,
-        )
+        begin
+          Cask::Upgrade.upgrade_casks!(
+            *casks,
+            force:               args.force?,
+            greedy:              args.greedy?,
+            greedy_latest:       args.greedy_latest?,
+            greedy_auto_updates: args.greedy_auto_updates?,
+            dry_run:             args.dry_run?,
+            binaries:            args.binaries?,
+            quarantine:          args.quarantine?,
+            require_sha:         args.require_sha?,
+            skip_cask_deps:      args.skip_cask_deps?,
+            verbose:             args.verbose?,
+            quiet:               args.quiet?,
+            args:,
+          )
+        rescue Timeout::Error => e
+          # When multiple casks are processed, `upgrade_casks!` handles each sequentially
+          # Skips are recorded inside the installer loop where errors are rescued
+          opoo "Timed out waiting for user input while upgrading casks. Some items may be skipped."
```

Comment:
Similarly here: having this inside `Cask::Upgrade` or `Cask::Installer` feels like it may make more sense to avoid repetition being needed.